### PR TITLE
bpo-40275: Add os_helper submodule in test.support

### PR DESCRIPTION
**Pull request by shihai1991**
Import the following vars, functions, classes in os_helper:
* rmtree()
* unlink()
* rmdir()
* FS_NONASCII
* TESTFN_UNICODE, TESTFN_NONASCII, TESTFN_UNENCODABLE, TESTFN_UNDECODABLE
* SAVEDCWD 
* temp_dir()
* change_cwd()
* temp_cwd()
* temp_umask()
* create_empty_file()
* make_bad_fd()
* EnvironmentVarGuard
* can_symlink(), skip_unless_symlink()
* can_xattr(), skip_unless_xattr()
* fs_is_case_insensitive()
* findfile
* fd_count()
* FakePath

https://bugs.python.org/issue40275

### CODE REVIEW

**Comment by shihai1991** [comment]
@vstinner Hi, victor. Pls have a look again when you have free time.

**Comment by vstinner** [comment]
For the first PR (this one), please leave `support/__init__.py` unchanged: only import names in os_helper. Also, please leave test_xxx.py unchanged.

**Comment by vstinner** [comment]
For example, os_helper.py would contain something like:

```
from test.support import TESTFN
...
```

**Comment by shihai1991** [comment]
> For example, os_helper.py would contain something like:
> 
> ```
> from test.support import TESTFN
> ...
> ```

Oh, My logic is the opposite.  IMHO, it is same behavior, no?

**Comment by vstinner** [comment]
Brett and me explained to you that a PR modifying 100+ files is a bad idea.

Adding a new submodule which only imports existing functions avoids modifying any existing test and so it becomes possible to convert tests slowly, one by one, or 20 by 20.

**Comment by shihai1991** [comment]
> Brett and me explained to you that a PR modifying 100+ files is a bad idea.
Pls forgive me, I have relealized that it's not easy to review and noisy too many developers.

> Adding a new submodule which only imports existing functions avoids modifying any existing test and so it becomes possible to convert tests slowly, one by one, or 20 by 20.

IMO: `a new submodule which only imports existing functions` same as `move existing functions in new submodule and reimport those existing functions in test.support`

Oh, wait. You are right. Some testcase use inner functions of test.support will be affected.

File: Doc/library/test.rst [hunk 1]
```diff
@@ -1634,3 +1452,197 @@ The :mod:`test.support.threading_helper` module provides support for threading t
...
+
+
+.. data:: FS_NONASCII
+
+   A non-ASCII character encodable by :func:`os.fsencode`.
+
+
+.. data:: SAVEDCWD
+
+   Set to :func:`os.getcwd`.
```
Comment (vstinner) [inline Doc/library/test.rst:1475]: Side note: I'm not sure why it exists in test.support. It seems like only test.libregrtest uses it. Maybe it should be moved (later) to test.libregrtest.

**Review by vstinner — COMMENTED** [review]
> move existing functions in new submodule and reimport those existing functions in test.support

Right. I tried to check if there are remaining functions in support/__init__.py which should be moved to os_helper. But since functions are not moved, it's not easy.

Would you mind to create a second PR which moves functions but adds aliases in support/__init__.py? It sounds like an even better approach.

**Comment by vstinner** [comment]
IMO PR #20765 is the right way to fix this issue.

File: Doc/library/test.rst (hunk 1 above)
Comment (shihai1991) [reply Doc/library/test.rst:1475]: Hm, I checked this detail. Looks like the describe info is right. It points that `SAVEDCWD=os.getcwd()`

**Pull request closed without merging**